src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/CaseHearingLocationService.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.sscs.service.hmc.topic;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.exception.InvalidHearingDataException;
+import uk.gov.hmcts.reform.sscs.exception.InvalidMappingException;
+import uk.gov.hmcts.reform.sscs.model.hmc.message.HmcMessage;
+import uk.gov.hmcts.reform.sscs.service.VenueService;
+
+import javax.validation.Valid;
+
+import static uk.gov.hmcts.reform.sscs.helper.service.CaseHearingLocationHelper.findVenue;
+import static uk.gov.hmcts.reform.sscs.helper.service.CaseHearingLocationHelper.getHearingFromCaseData;
+import static uk.gov.hmcts.reform.sscs.helper.service.CaseHearingLocationHelper.updateHearing;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CaseHearingLocationService {
+
+    private final VenueService venueService;
+
+    public void updateVenue(HmcMessage hmcMessage, @Valid SscsCaseData sscsCaseData)
+            throws InvalidHearingDataException, InvalidMappingException {
+
+        Hearing oldHearing = getHearingFromCaseData(hmcMessage, sscsCaseData);
+
+        String updatedVenueId = hmcMessage.getHearingUpdate().getHearingVenueId();
+        Venue venue = findVenue(updatedVenueId, venueService);
```

Comment:
minor / (major to me): we shouldn't really be using a service this way - can you get rid of this method and extract the logic into here instead? Then call `mapVenueDetailsToVenue` from here. You'll probably need to update some tests though.